apps/client/src/features/rundown/Rundown.tsx

```diff
@@ -10,39 +10,81 @@ import {
   RundownCached,
   SupportedEvent,
 } from 'ontime-types';
-import { getFirstNormal, getNextNormal, getPreviousNormal } from 'ontime-utils';
+import { getFirstNormal, getLastNormal, getNextNormal, getPreviousNormal } from 'ontime-utils';
 
 import { useEventAction } from '../../common/hooks/useEventAction';
 import useFollowComponent from '../../common/hooks/useFollowComponent';
 import { useRundownEditor } from '../../common/hooks/useSocket';
 import { AppMode, useAppMode } from '../../common/stores/appModeStore';
 import { useEditorSettings } from '../../common/stores/editorSettings';
-import { cloneEvent } from '../../common/utils/eventsManager';
+import { useEventCopy } from '../../common/stores/eventCopySore';
+import { cloneEntry, cloneEvent } from '../../common/utils/eventsManager';
 
 import QuickAddBlock from './quick-add-block/QuickAddBlock';
 import RundownEmpty from './RundownEmpty';
+import { useEventSelection } from './useEventSelection';
 
 import style from './Rundown.module.scss';
 
 const RundownEntry = lazy(() => import('./RundownEntry'));
 
+type KeyCombo = { altKey: boolean; ctrlKey: boolean; shiftKey: boolean; code: string };
+const checkKeys = (event: KeyboardEvent, keyCombo: KeyCombo) => {
+  return (
+    keyCombo.altKey === event.altKey &&
+    keyCombo.ctrlKey === event.ctrlKey &&
+    keyCombo.shiftKey === event.shiftKey &&
+    keyCombo.code === event.code
+  );
+};
+
+const keyCombos = {
+  selectUp: (event: KeyboardEvent) =>
+    checkKeys(event, { altKey: true, ctrlKey: false, shiftKey: false, code: 'ArrowUp' }),
```

Comment:
I think the sending if modifiers can be improved

How about an array of key codes?
Checkout react-hotkeys for some inspiration

Reply:
yeah I did find that one
but it looks to me like they are manually maintaining an array of pressed keys...
right now I'm managing with only using the KeybordEvent
I would need some help implementing something different